### PR TITLE
feat: add registration based drawing for a more manageable draw system

### LINKED ISSUE #7: Update the Drawing Process to Use Registry
Right now, adding new Drawables requires some modifications of the MediaPlayer object. I don't like that since I want others to be able to build from this. Therefore, I believe adding a registry flow is the way to go.

Action Items:
- Gut MediaPlayer.Animate() and create a draw_registry variable to keep track of all Drawables that want to be drawn.
- In MediaPlayer.Animate(), run through the registry and just animate whatever is there. Deletion is also simpler here.

### PR DESCRIPTION
**Pull request by QriusDev**

### CODE REVIEW

File: widgets/qp_canvas.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
+/**
+ * This class defines a canvas and how it will draw its components
+ */
+class QPCanvas
+{
+    handle = undefined;
+    width = 0;// CONSTANTS
```
Comment (dylta) [inline widgets/qp_canvas.js]: Can you fix the spacing of the comments on here and line 9?

File: player.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /**
  * An media player
  */
-class MediaPlayer 
+class MediaPlayer extends QPCanvas
```
Comment (dylta) [inline player.js:4]: Do we plan on using multiple inheritance when MediaMap is implemented?

File: player.js [hunk 3]
```diff
@@ -107,22 +131,30 @@ class MediaPlayer
                 10
             );
             this.timeline.color = "#b1b1b1";
-            this.timeline.draw();
+        });
 
-            // Seeker Tail
+        // Seeker Tail
+        // HACK: until we have the seeker tail checked in
```
Comment (dylta) [inline player.js]: Can you add a TODO to remove this whenever we can

File: player.js (hunk 2 above)
Comment (QriusDev) [reply player.js:4]: Not sure if you tagged the right line but the **MediaMap** would be in place of the **media** object that the **MediaPlayer** holds as a property.

File: widgets/qp_canvas.js (hunk 1 above)
Comment (QriusDev) [reply widgets/qp_canvas.js]: I'm not even sure why that was there in the first place lmao.

File: player.js (hunk 2 above)
Comment (dylta) [reply player.js:4]: Oh yeah duh lol

**Review by dylta — APPROVED** [review]

**Pull request merged**